requirements-dev.txt

```diff
@@ -0,0 +1,3 @@
+# Git commit hook handler.
+# SEE: https://pre-commit.com/
+pre-commit
```

Comment:
Can you pin the version? I've verified that the most recent version (4.0.1) is working

Reply:
@Nerogar We can't pin the version, but we can do some things.

- Pre-commit is installed user-wide. A single installation shared by every Git repo that uses it. It places itself in the user's "Python binaries" PATH, where other "pip-installed system-wide binaries" live.
- Its configuration (database, repos etc) are stored in `~/.cache/pre-commit`. By setting `== 4.0.1` we would downgrade people's pre-commit and could break their system.
- It is just a framework. The actual checks where we care about versions are in our configuration file.
- They follow semantic versioning. When they raise the major version, it's because they deprecated some feature, but it's always less-used features. We use a very standardized, basic set of its features.
- Every time they deprecate something, they make their tool loudly shout about it when you try to use that feature. So we can't miss any changes.

All this being said, we know we use the feature-set as of 4.0.1, so we could force anyone's old installations to upgrade to that version or higher, by saying `pre-commit >= 4.0.1`. I'll make that change.